Shared/Notifications/NotificationsManager.swift

```diff
@@ -0,0 +1,193 @@
+    func verifyCurrentNotifications() {
+        populateScheduledNotifications()
+        updateNotifications(menuItemNames: Defaults[\.favoriteMenuItems])
+    }
+    
+    /// Schedule a notification for a menu item being served at any of the eateries in eateryDisplayNames, to be triggered on date
+    private func setUpNotification(menuItemName: String, eateryDisplayNames: [String], on date: Date, eventName: CampusEatery.EventName) {
+        guard date > Date(), !eateryDisplayNames.isEmpty else { return }
+        
+        let notifContent = UNMutableNotificationContent()
+        notifContent.title = self.favoriteMenuItemNotifTitle
+        var multipleServingsPostfix: String
+        switch eateryDisplayNames.count {
+        case 1:
+            multipleServingsPostfix = ""
+        case 2:
+            multipleServingsPostfix = " and \(eateryDisplayNames[1])"
+        case 3:
+            multipleServingsPostfix = ", \(eateryDisplayNames[1]), and 1 other"
+        default:
+            multipleServingsPostfix = ", \(eateryDisplayNames[1]), and \(eateryDisplayNames.count - 2) others"
+        }
+        notifContent.body = "\(menuItemName) is being served at \(eateryDisplayNames[0])" + multipleServingsPostfix
+        notifContent.sound = UNNotificationSound.default()
+        notifContent.userInfo = [
+            notifContentDateKey : date,
+            notifContentEateriesKey : eateryDisplayNames,
+            notifContentEventNameKey: eventName,
+            notifContentMenuItemKey : menuItemName
+        ]
+
+        let notifDateComponents = Calendar.current.dateComponents([.second, .minute, .hour, .day, .month, .year], from: date)
+        let notifTrigger = UNCalendarNotificationTrigger(dateMatching: notifDateComponents, repeats: false)
+        let notifRequest = UNNotificationRequest(identifier: UUID().uuidString, content: notifContent, trigger: notifTrigger)
+        UNUserNotificationCenter.current().add(notifRequest, withCompletionHandler: nil)
+        
+        var newMenuItemNotifRequests: [UNNotificationRequest]
+        if let menuItemNotifRequests = scheduledNotifications[menuItemName] {
+            newMenuItemNotifRequests = menuItemNotifRequests
+        } else {
+            newMenuItemNotifRequests = [UNNotificationRequest]()
+        }
+        newMenuItemNotifRequests.append(notifRequest)
+        scheduledNotifications[menuItemName] = newMenuItemNotifRequests
+    }
+    
+    // Searches through known events and sets up a notification for a meal event if one or multiple dining halls are serving any of the menu items in menuItemNames
+    func updateNotifications(menuItemNames: [String]) {
+        NetworkManager.shared.getCampusEateries(useCachedData: true) { (eateries, error) in
+            guard let eateries = eateries else { return }
+            
+            var itemServingsByMealAndDate = [String : [Int : [(eatery: String, eventStart: Date, eventName: CampusEatery.EventName)]]]()
```

Comment:
is there anyway we can simplify or break down the logic from line 101-168 (separate into different helper functions // add comments)? it just seems a bit daunting to try to understand